conan/tools/premake/premake.py

```diff
@@ -1,39 +1,184 @@
+from conan.errors import ConanInvalidConfiguration, ConanException
+from conan.tools.microsoft import MSBuild
+from conan.tools.apple import XcodeBuild
+
+import glob
+import os.path
+
+# Dictionary for version to year lookups
+PREMAKE_VS_VERSION = {
+    # Visual Studio -> Year
+    # https://en.wikipedia.org/wiki/Visual_Studio#History
+    '8'     :   '2005',
+    '9'     :   '2008',
+    '10'    :   '2010',
+    '11'    :   '2012',
+    '12'    :   '2013',
+    '14'    :   '2015',
+    '15'    :   '2017',
+    '16'    :   '2019',
+    '17'    :   '2022',
+
+    # MSVC -> Year 
+    # https://en.wikipedia.org/wiki/Microsoft_Visual_C%2B%2B#Internal_version_numbering
+    '14.0'  :   '2005',
+    '15.0'  :   '2008',
+    '16.0'  :   '2010',
+    '17.0'  :   '2012',
+    '18.0'  :   '2013',
+    '19.0'  :   '2015',
+    '19.1'  :   '2017',
+    '19.10' :   '2017',
+    '19.11' :   '2017',
+    '19.12' :   '2017',
+    '19.13' :   '2017',
+    '19.14' :   '2017',
+    '19.16' :   '2017',
+    '19.2'  :   '2019',
+    '19.20' :   '2019',
+    '19.21' :   '2019',
+    '19.22' :   '2019',
+    '19.23' :   '2019',
+    '19.24' :   '2019',
+    '19.25' :   '2019',
+    '19.26' :   '2019',
+    '19.27' :   '2019',
+    '19.28' :   '2019',
+    '19.29' :   '2019',
+    '19.3'  :   '2022',
```

Comment:
We might want to drop all this part, and just use the MSVC without dot, and process the input to simplify it to 191, 192, etc.